### PR TITLE
PLAT-847: ZenCard component

### PR DESCRIPTION
**Pull request by mplesec**
[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/PLAT-847)

#### Description
PR adds ZenCard component.

#### ChangeLOG

- [ ] I have added all notable deployment/development environment (onprem!) changes to [CHANGELOG.md](https://github.com/reciprocity/zen-ui/blob/main/README.md)

#### Useful links
[Zen UI readme](https://github.com/reciprocity/zen-ui/blob/main/README.md)

[PR review guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/pull_request_reviews.md)

[Git commit guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/commit_guidelines.md)

### CODE REVIEW

File: src/components/zen-card/zen-card.scss [hunk 1]
```diff
@@ -0,0 +1,19 @@
+@import '../../zen-styles/component-helpers.scss';
+
+:host {
+  display: block;
+
+  border: 1px solid #ced4da;
```
Comment (f-jauregui) [inline src/components/zen-card/zen-card.scss:6]: Change to `1px solid $color-gray-200;`

File: src/components/zen-card/zen-card.scss [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  cursor: pointer;
+  transition: box-shadow $transition-visibility;
```
Comment (f-jauregui) [inline src/components/zen-card/zen-card.scss:8]: Border should transition too (for hover).

File: src/components/zen-card/zen-card.scss [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+:host(:hover) {
+  box-shadow: 0 0 15px rgba(0, 0, 0, 0.1);
+}
```
Comment (f-jauregui) [inline src/components/zen-card/zen-card.scss:13]: Based on [this](https://www.figma.com/file/EQAOxhTU1yWyoXhvlLeyws/New-platform-POC-%7C-Shelley?version-id=561595729&node-id=1348%3A4257), on hover border is white or invisible.

File: src/components/zen-card/zen-card.scss [hunk 4]
```diff
@@ -0,0 +1,19 @@
+@import '../../zen-styles/component-helpers.scss';
+
+:host {
+  display: block;
+
```
Comment (f-jauregui) [inline src/components/zen-card/zen-card.scss:5]: Missing `border-radius: 2px;`

**Review by f-jauregui — CHANGES_REQUESTED** [review]

File: src/components/zen-card/zen-card.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  styleUrl: 'zen-card.scss',
+  shadow: true,
+})
+export class ZenCard {
+  /** Disables card. */
+  @Prop() readonly disabled = false;
+
+  render(): HTMLElement {
+    return (
+      <Host class={{ disabled: this.disabled }}>
```
Comment (chavlji) [inline src/components/zen-card/zen-card.tsx:14]: Don't mirror props to class. Use `{reflect: true}` on prop instead and css selector `:host([disabled]) {...}`

File: src/components/zen-card/zen-card.scss (hunk 2 above)
Comment (chavlji) [reply src/components/zen-card/zen-card.scss:8]: Do we even need tranition on hover? Cards are usually static.

File: src/components/zen-card/zen-card.scss [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  border: 1px solid #ced4da;
+  cursor: pointer;
+  transition: box-shadow $transition-visibility;
+}
+
+:host(:hover) {
+  box-shadow: 0 0 15px rgba(0, 0, 0, 0.1);
```
Comment (chavlji) [inline src/components/zen-card/zen-card.scss:12]: Is this the right shadow? Check shadows that designers provided:
https://www.figma.com/file/rWXEQiJcBXbkMY8V4NSM1d/ZenGRC-Component-Library?node-id=1463%3A13

File: src/components/zen-card/test/zen-card.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  it('renders', async () => {
+    const page = await newSpecPage({
+      components: [ZenCard],
+      html: `<zen-card>Content</zen-card>`,
+    });
+
+    expect(page.root.innerHTML).toEqual('Content');
+  });
+
+  it('applied `disabled` class if `disabled` prop is set', async () => {
```
Comment (chavlji) [inline src/components/zen-card/test/zen-card.spec.tsx:15]: You won't need this test if you don't mirror prop to class.

File: src/components/zen-card/zen-card.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+import data from '../../../stencilDocs.json';
+const compData = data.components.find(n => n.tag === 'zen-card');
+const argTypes = getArgTypes(compData);
+
+<Meta title="Containers/Card" component="zen-card" argTypes={argTypes} />
+
+export const Story1 = () => {
+  return html/*html*/ `
+    <zen-card id="card1">
+      <div style="padding: 1rem;">Put your content here 😎</div>
```
Comment (chavlji) [inline src/components/zen-card/zen-card.stories.mdx:16]: I'd avoid emojis. Looks a bit unprofessional. Add some zen-icon instead.

**Review by chavlji — CHANGES_REQUESTED** [review]
Nice. A few changes though...

File: src/components/zen-card/zen-card.scss (hunk 3 above)
Comment (mplesec) [reply src/components/zen-card/zen-card.scss:13]: Discussed offline. Won't be adding it now, it looks worse imo. If someone from design later asks for it, we'll discuss it then.

File: src/components/zen-card/zen-card.scss (hunk 2 above)
Comment (mplesec) [reply src/components/zen-card/zen-card.scss:8]: Usually, someone should happen when hovering over it.

File: src/components/zen-card/zen-card.scss (hunk 6 above)
Comment (mplesec) [reply src/components/zen-card/zen-card.scss:12]: I copied it over from a shadow in a different component. Suggest we make a ticket to unify these now that they're available.

File: src/components/zen-card/zen-card.tsx (hunk 5 above)
Comment (mplesec) [reply src/components/zen-card/zen-card.tsx:14]: Nice, very useful

File: src/components/zen-card/test/zen-card.spec.tsx (hunk 7 above)
Comment (mplesec) [reply src/components/zen-card/test/zen-card.spec.tsx:15]: Removed

File: src/components/zen-card/zen-card.stories.mdx (hunk 8 above)
Comment (mplesec) [reply src/components/zen-card/zen-card.stories.mdx:16]: Unless we agree to this, does it really matter since it's the content for a usage example?

**Review by f-jauregui — APPROVED** [review]

File: src/components/zen-card/zen-card.stories.mdx (hunk 8 above)
Comment (f-jauregui) [reply src/components/zen-card/zen-card.stories.mdx:16]: Agree with @chavlji 😅

Comment (chavlji) [reply src/components/zen-card/zen-card.stories.mdx:16]: It's content, true. But content also builds up a Storybook that is our product and should look professional.
@f-jauregui , @mihaelpalfi your call.

File: src/components/zen-card/zen-card.scss (hunk 6 above)
Comment (chavlji) [reply src/components/zen-card/zen-card.scss:13]: 👍🏻 https://reciprocitylabs.atlassian.net/browse/PLAT-1049

**Review by chavlji — APPROVED** [review]

File: src/components/zen-card/zen-card.stories.mdx (hunk 8 above)
Comment (mplesec) [reply src/components/zen-card/zen-card.stories.mdx:16]: Hah sure, tho I disagree with since since emojis add flare 🔥, I'll remove them.

**Pull request merged**